test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs

```diff
@@ -0,0 +1,310 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using ComparisonExtensions;
+using FluentAssertions;
+using Xunit;
+
+namespace ComparisonExtensionsTests
+{
+    public class DoubleComparisonExtensionsTest
+    {
+        public record DoubleEqualityScenario(double Left, double Right, string Name = null)
+        {
+            public override string ToString() => this.Name == null
+                ? $"left = {this.Left}, right = {this.Right}"
+                : this.Name;
+        }
+
+        public record DoubleZeroScenario(double Value, bool ExpectZero, string Name = null)
+        {
+            public override string ToString() => this.Name == null
+                ? $"value = {this.Value}, expectZero = {this.ExpectZero}"
+                : this.Name;
+        }
+
+        public record DoubleComparisonScenario(double Left, double Right, bool LeftIsGreaterThanRight, string Name = null)
+        {
+            public override string ToString() => this.Name == null
+                ? $"left = {this.Left}, right = {this.Right}, leftIsGreaterThanRight = {this.LeftIsGreaterThanRight}"
+                : this.Name;
+        }
+
+        public record DoubleNoComparisonScenario(double Left, double Right, string Name = null)
+        {
+            public override string ToString() => this.Name == null
+                ? $"left = {this.Left}, right = {this.Right}"
+                : this.Name;
+        }
+
+        public static IEnumerable<object[]> EqualTestCasesData => EqualTestCases.Select(testCase => new object[] { testCase });
+        private static IEnumerable<DoubleEqualityScenario> EqualTestCases
+        {
+            get
+            {
+                yield return new(0.1d, 0.1d);
+                yield return new(-0.1d, -0.1d);
+                yield return new(0.000000001d, 0.000000001d);
+                yield return new(double.PositiveInfinity, double.PositiveInfinity);
+                yield return new(double.NegativeInfinity, double.NegativeInfinity);
+                yield return new(double.Epsilon, 0.0d);
+                yield return new(0.0d, double.Epsilon);
+
+                yield return new((double)0.1 + (double)0.2 + (double)0.7, 1, "works well with integers");
```

Comment:
casting to `double` is redundant here.

Reply:
it is, I was trying to leave it how the original test was but it's updated now